### PR TITLE
Assignment 3

### PR DESCRIPTION
**Pull request by ericlplanteschool**
implemented base requirements

### CODE REVIEW

**Comment by mmwoodfo** [comment]
You're making this PR from a private repo - So I can't open your code base which means I can't give you points for the unit tests running or the code compiling.

File: CanHazFunny/CanHazFunny/CanHazFunny.csproj [hunk 1]
```diff
@@ -3,6 +3,11 @@
   <PropertyGroup>
     <OutputType>Exe</OutputType>
     <TargetFramework>net6.0</TargetFramework>
+    <LangVersion>10.0</LangVersion>
+    <Nullability>enable</Nullability>
+    <CodeAnalysisTreatWarningsAsErrors>true</CodeAnalysisTreatWarningsAsErrors>
+    <EnableNETAnalyzers>true</EnableNETAnalyzers>
+    <TreatWarningsAsErrors>true</TreatWarningsAsErrors>
```
Comment (mmwoodfo) [inline CanHazFunny/CanHazFunny/CanHazFunny.csproj:10]: Make sure to enable these in the test project as well

File: CanHazFunny/CanHazFunny.Tests/JesterTests.cs [hunk 2]
```diff
@@ -5,6 +5,6 @@ namespace CanHazFunny.Tests
     [TestClass]
     public class JesterTests
     {
-        
+
```
Comment (mmwoodfo) [inline CanHazFunny/CanHazFunny.Tests/JesterTests.cs:8]: There should be tests here.

File: CanHazFunny/CanHazFunny/Interfaces.cs [hunk 3]
```diff
@@ -0,0 +1,9 @@
+namespace CanHazFunny{
+    public interface IGetJoke{
+        string GetJoke();
+    }
+
+    public interface IPrintJoke{
+        void PrintJoke(string arg);
+    }
+}
```
Comment (mmwoodfo) [inline CanHazFunny/CanHazFunny/Interfaces.cs:9]: You should split these into two separate files. It's bad practice to keep all of your interfaces or classes in a single file.

File: CanHazFunny/CanHazFunny/PrintService.cs [hunk 4]
```diff
@@ -0,0 +1,11 @@
+using System;
+
+namespace CanHazFunny{
+
+    public class PrintService : IPrintJoke 
+    {
+        public void PrintJoke(string joke){
```
Comment (mmwoodfo) [inline CanHazFunny/CanHazFunny/PrintService.cs:7]: Keep these argument names the same. Your interface calls the parameter `args`

File: CanHazFunny/CanHazFunny/Jester.cs [hunk 5]
```diff
@@ -0,0 +1,32 @@
...
+
+        private IPrintJoke printer;
+
+        public Jester(IGetJoke Joker, IPrintJoke Printer)
+        {
+        joker = Joker;
+
+        printer = Printer;
+
+        if(joker == null || printer == null){
```
Comment (mmwoodfo) [inline CanHazFunny/CanHazFunny/Jester.cs:17]: You should check for null before setting the properties

File: CanHazFunny/CanHazFunny/Jester.cs [hunk 6]
```diff
@@ -0,0 +1,32 @@
...
+            throw new ArgumentException("null dependency");
```
Comment (mmwoodfo) [inline CanHazFunny/CanHazFunny/Jester.cs:18]: this should be an argument null exception and you should specify which one is null

**Comment by ericlplanteschool** [comment]
> You're making this PR from a private repo - So I can't open your code base which means I can't give you points for the unit tests running or the code compiling.

That is odd my repository is set to public

**Comment by mmwoodfo** [comment]
image
Perhaps this is the issue, I thought it was private, but looks like it's deleted?

**Comment by ericlplanteschool** [comment]
> image Perhaps this is the issue, I thought it was private, but looks like it's deleted?

I had to recreate my fork because for whatever reason it wasn't puling updates when i tried to fetch from the head, didn't realize it would delete any pending PRs

**Pull request closed without merging**